engine/class_modules/priest/sc_priest_discipline.cpp

```diff
@@ -132,68 +132,77 @@ struct penance_t final : public priest_spell_t
 
 struct power_word_solace_t final : public priest_spell_t
 {
-  power_word_solace_t( priest_t& player, util::string_view options_str )
-    : priest_spell_t( "power_word_solace", player, player.talents.power_word_solace )
+  power_word_solace_t( priest_t& p, util::string_view options_str )
+    : priest_spell_t( "power_word_solace", p, p.talents.discipline.power_word_solace )
   {
     parse_options( options_str );
-
     cooldown->hasted = true;
-
-    travel_speed = 0.0;  // DBC has default travel taking 54seconds.....
+    travel_speed     = 0.0;  // DBC has default travel taking 54 seconds
   }
 
   void impact( action_state_t* s ) override
   {
     priest_spell_t::impact( s );
-
     double amount = data().effectN( 2 ).percent() / 100.0 * priest().resources.max[ RESOURCE_MANA ];
     priest().resource_gain( RESOURCE_MANA, amount, priest().gains.power_word_solace );
   }
 };
 
+// Purge the wicked
 struct purge_the_wicked_t final : public priest_spell_t
 {
-  struct purge_the_wicked_dot_t final : public priest_spell_t
+  purge_the_wicked_t( priest_t& p ) : priest_spell_t( "purge_the_wicked", p, p.talents.discipline.purge_the_wicked )
   {
-    purge_the_wicked_dot_t( priest_t& p )
-      : priest_spell_t( "purge_the_wicked", p, p.talents.purge_the_wicked->effectN( 2 ).trigger() )
-    {
-      may_crit = true;
-      // tick_zero = false;
-      energize_type = action_energize::NONE;  // disable resource generation from spell data
-      background    = true;
+    may_crit  = true;
+    tick_zero = false;
 
-      apply_affecting_aura( priest().talents.throes_of_pain );
-    }
+    // Throes of Pain: Spell Direct and Periodic 3%/5% gain
+    apply_affecting_aura( priest().talents.throes_of_pain );
+  }
 
-    void tick( dot_t* d ) override
-    {
-      priest_spell_t::tick( d );
+  purge_the_wicked_t( priest_t& p, util::string_view options_str ) : purge_the_wicked_t( p )
+  {
+    parse_options( options_str );
+  }
+
+  void trigger( player_t* target )
+  {
+    background = true;
+    player->sim->print_debug( "{} triggered purge_the_wicked on target {}.", priest(), *target );
+
+    set_target( target );
+    execute();
+  }
+
+  void impact( action_state_t* s ) override
+  {
+    priest_spell_t::impact( s );
 
-      if ( d->state->result_amount > 0 )
-      {
-        trigger_power_of_the_dark_side();
-      }
+    if ( result_is_hit( s->result ) )
+    {
     }
-  };
+  }
 
-  purge_the_wicked_t( priest_t& p, util::string_view options_str )
-    : priest_spell_t( "purge_the_wicked", p, p.talents.purge_the_wicked )
+  timespan_t tick_time( const action_state_t* state ) const override
   {
-    parse_options( options_str );
+    timespan_t t = priest_spell_t::tick_time( state );
+    return t;
+  }
```

Comment:
this block isnt actually doing anything to `t` so can be removed

Reply:
Removed